File: 3621-minimum-operations-to-make-array-values-equal-to-k/minimum-operations-to-make-array-values-equal-to-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minOperations = function (nums, k) {
    const seen = new Set(nums);
    let count = 0;

    for (const num of seen) {
        if (num < k) {
            return -1;
        }
        else if (num > k) {
            count += 1;
        }
    }

    return count;
};

File: 3621-minimum-operations-to-make-array-values-equal-to-k/minimum-operations-to-make-array-values-equal-to-k.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        unordered_set<int> seen(nums.begin(), nums.end());
        int count = 0;

        for (auto num : seen) {
            if (num < k) {
                return -1;
            }
            if (num > k) {
                count++;
            }
        }

        return count;
    }
};